### ISSUE TITLE
Is this library abandonware?

### ISSUE THREAD

**Issue by sustained** [Patreon/patreon-php#55]
The last commit and last official response to an issue was late August.

Has this library been abandoned by Patreon, or are the authors just busy at this time?

If it's not abandoned, will it ever support v2 of the API?

Also, how about documentation? Getting this working has been quite a pain. Especially with that annoyingly hard to work with JSON library you're using. It's good that we can disable the parsing, though - but it would be nice if that (among other things) was documented.

**Comment by nperickson** [comment]
Hey @sustained -

Not abandoned, just a low priority because of time right now. 

We have updating, and fundamentally improving, all of our client libraries to work with v2 of the API in the backlog. Documentation around the client libraries is among the things we'd like to include in the "improve" section.

**Comment by sustained** [comment]
Good to hear! I could potentially help write some documentation but I'd honestly rather wait for a v2-based API before doing so?

**Comment by furai** [comment]
Any ETA on v2 support?

**Comment by codebard** [comment]
Yeah, its almost ready. I'd say next week or the week after that it should be tested and released, depending on the decision of Platform team.

**Comment by furai** [comment]
And what if I don't want to wait and can deal with bugs myself? Can I get somehow the upcoming new version? Both with removed art4 json lib and v2 support, even if its buggy?

**Comment by codebard** [comment]
Today i will be merging branches into my fork's master. After that you can grab the master and just use it i believe. But naturally when it is released via packagist, you would need to move onto the package from packagist if you are using composer. Or alternatively you can just maintain it from Patreon's github master in future.

**Comment by codebard** [comment]
Fork is finalized and ready. After this only minor corrections may come. 

https://github.com/codebard/patreon-php

**Comment by ph-il** [comment]
Any news... Where like 2 month laters

**Comment by furai** [comment]
From what I can see code in master already uses v2? Am I missing something here?

**Comment by codebard** [comment]
You are correct - code in master is already completed, tested. Its just not released at packagist or here. You can use it.

**Comment by furai** [comment]
And I have been using it for nearly 2 months now. :) Thanks again for working on it.

**Comment by codebard** [comment]
Great to hear. So you had no issues with it then?

**Comment by furai** [comment]
No, not really. But I needed just to get the patreon user ID.

**Comment by codebard** [comment]
Yeah, there isnt any premade/ready function for getting user ids, creator ids or pledges etc.

**Comment by herrvigg** [comment]
Patreon PHP library 0.3.x for  API v1.
Patreon PHP library 1.0.0 for  API v2. Not compatible with v1. Not released to packagist.

Is that even possible to create more confusions?! 😕
Imo at least the minor version of the library should follow the API version, if not the major. It's just basic common sense.

**Comment by codebard** [comment]
API v1 is going to be deprecated at a certain point in time in near future. Hence the github being only for api v2. API v2 is not compatible with API v1. 

One of the reasons that this latest version is not released at packagist yet is because there are a noticeable number of apps using the packagist version, and if they just update to this latest version of the lib they will lose backwards compatibility. Hence we need to arrange a smooth transition for them to API v2 before pushing the latest version and breaking backwards compatibility. That's the reason why this version is not at packagist yet.

TL:DR; You can just clone this github repo to your app and use it, and you will be set with API v2 and backwards compatible into foreseeable future.

**Comment by soatok** [comment]
As a result of the discussion in #69, I am maintaining a fork instead:

https://github.com/soatok/patreon-php

Feel free to use it. I'll have a fast turnaround on any bugs reported. (Most likely, fixed within 24 hours.)
